server/src/main/java/io/druid/segment/realtime/appenderator/FiniteAppenderatorDriver.java

```diff
@@ -382,21 +432,30 @@ private SegmentIdentifier getSegment(final DateTime timestamp, final String sequ
   /**
    * Move a set of identifiers out from "active", making way for newer segments.
    */
-  private void moveSegmentOut(final String sequenceName, final List<SegmentIdentifier> identifiers)
+  private boolean moveSegmentOut(
+      final String sequenceName,
+      final List<SegmentIdentifier> identifiers,
+      boolean ignoreAbsent
```

Comment:
When moving segments out after publishing them, some of the published segments may be already moved out from activeSegments due to the ```maxRowsPerSegment``` limit. I added a comment.